setup.py

```diff
@@ -0,0 +1,40 @@
+import os
+from typing import List
+
+from setuptools import setup
+
+ROOT_DIR = os.path.dirname(__file__)
+
+
+def get_path(*filepath) -> str:
+    return os.path.join(ROOT_DIR, *filepath)
+
+
+def get_requirements() -> List[str]:
+    """Get Python package dependencies from requirements.txt."""
+
+    def _read_requirements(filename: str) -> List[str]:
+        with open(get_path(filename)) as f:
+            requirements = f.read().strip().split("\n")
+        resolved_requirements = []
+        for line in requirements:
+            if line.startswith("-r"):
+                resolved_requirements += _read_requirements(line.split()[1])
+            elif line.startswith("--"):
+                continue
+            else:
+                resolved_requirements.append(line)
+        return resolved_requirements
```

Comment:
Is all this custom logic really required for parsing the requirements file? I thought were standard libraries for doing this stuff.

Reply:
mmh I did a quick search and couldn't find a standard library to parse the `requirements.txt` file. I took that from vllm code directly, and It seems that the standard way to install with `setup.py` is using `install_requires` instead of using `requirements.txt` (it is not common to mix `setup.py` and `requirements.txt`). I seems that one previously could use the function for pip for parsing: `pip.req.parse_requirements`, but this is not recommended because it is for pip internal use. [See discussion](https://stackoverflow.com/questions/14399534/reference-requirements-txt-for-the-install-requires-kwarg-in-setuptools-setup-py).